File: problem_0002/problem_0002.py
#! env python

def F(limit=float('Inf')):
  a=0
  b=1
  while a+b<limit:
   if (a+b)%2==0: yield a+b
   a,b = b,a+b

f = F(4*10**6)
print sum(f)

File: problem_0002/problem_0002.java
class problem_0002
{
  public static void main(String args[]){
    int sum = 0;
    int a = 0;
    int b = 1;
    while (b < 4000000){
      if (b%2 == 0){
        sum += b;
      }
      b = b + a;
      a = b - a;
    }
 
    System.out.println(sum);
  }
}
